pq-crypto/EVP_kem.c

```diff
@@ -0,0 +1,130 @@
+/********************************************************************************************
+* KEM Library
+*
+* Abstract: Post-Quantum Algorithm API
+*********************************************************************************************/
+
+#include <stdint.h>
+#include <stdlib.h>
+#include "EVP_kem.h"
+#include "EVP_pq_constants.h"
+
+struct pq_kem evp_sike_p434_r3 = {
+        .name = "SIKEp434r3-KEM",
+        .public_key_length = SIKE_P434_R3_PUBLIC_KEY_BYTES,
+        .private_key_length = SIKE_P434_R3_PRIVATE_KEY_BYTES,
+        .shared_secret_key_length = SIKE_P434_R3_SHARED_SECRET_BYTES,
+        .ciphertext_length = SIKE_P434_R3_CIPHERTEXT_BYTES,
+        .generate_keypair = &evp_sike_p434_r3_crypto_kem_keypair,
+        .encapsulate = &evp_sike_p434_r3_crypto_kem_enc,
+        .decapsulate = &evp_sike_p434_r3_crypto_kem_dec,
+};
+
+const struct pq_kem_params evp_sike_kem_params = {
+    .kem = &evp_sike_p434_r3,
```

Comment:
Is this necessary/valuable? I don't think we necessarily want to have a different `struct` type for each new KEM that we add. Maybe `pq_kem_params_alloc` should take a `pq_kem *` argument along with the `pq_kem_params *` argument and just make the `.kem = kem` assignment in that function before performing the `malloc`s. 

My thought is that I don't think we'll want to create a unique, hardcoded `pq_kem_params` for each new KEM we add.